rlclientlib/model_mgmt/file_model_loader.cc

```diff
@@ -0,0 +1,79 @@
+#include "file_model_loader.h"
+#include "err_constants.h"
+#include "api_status.h"
+#include <fstream>
+#include <utility>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#ifndef _WIN32
+#include <unistd.h>
+#endif
+
+#ifdef _WIN32
+#define stat _stat
+#endif
+
+namespace reinforcement_learning { namespace model_management {
+
+  int file_model_loader::get_file_modified_time(time_t& file_time, api_status* status) const {
+    struct stat result {};
+    if (stat(_file_name.c_str(), &result) == 0)
+    {
+      file_time = result.st_mtime;
+      return error_code::success;
+    }
+    RETURN_ERROR_LS(_trace, status, file_stats_error) << " file_name = " << _file_name;
+  }
+  
+  file_model_loader::file_model_loader(std::string file_name, bool file_must_exist, i_trace* trace_logger)
+    : _file_name{ std::move(file_name) }, _file_must_exist{ file_must_exist }, _trace{ trace_logger }
+  {}
+
+  int file_model_loader::init(api_status* status) {
+    if (_file_must_exist) {
+      std::ifstream in_strm(_file_name.c_str(), std::ios::in | std::ios::binary | std::ios::ate);
+      if (!in_strm.good()) {
+        RETURN_ERROR_LS(_trace, status, file_open_error) << " file_name = " << _file_name;
+      }
+    }
+    return error_code::success;
+  }
+
+  int file_model_loader::get_data(model_data& data, api_status* status) {
+
+    std::ifstream in_strm(_file_name.c_str(), std::ios::in|std::ios::binary|std::ios::ate);
```

Comment:
How costly is opening the file? Perhaps we should check if it was modified prior to opening it?

Reply:
it is done on higher layer - see get_file_modified_time above,   which is called before

Reply:
But in the sequence of the get_data function, it is called after creating the ifstream object though?

Reply:
oh yes you right

Reply:
Does not happen in the hot path.   Not clear what the perf impact is.